service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java

```diff
@@ -504,6 +505,11 @@ private DeviceTwinDevice jsonToDeviceTwinDevice(String json) throws IOException
         deviceTwinDevice.setConnectionState(twinState.getConnectionState());
         deviceTwinDevice.setConfigurations(twinState.getConfigurations());
 
+        if (twinState.getModelId() != null && !twinState.getModelId().isEmpty())
+        {
```

Comment:
Same comment here

Reply:
I would like to understand if there is a reasoning behind why should we set model id when null/empty?

Reply:
To avoid SDK's having any check. The point I was making is we should leave it to service to throw incase there is no model id and not customize it here.

Reply:
ok. done